src/requirements.ts

```diff
@@ -29,30 +34,18 @@ interface ErrorData {
   message: string;
   label: string;
   command: string;
-  commandParam: any;
+  commandParam: string | vscode.Uri;
 }
 
-export async function resolveRequirements(): Promise<RequirementsData> {
-  const javaHome = await checkJavaRuntime();
+export async function resolveRequirements(context: vscode.ExtensionContext): Promise<RequirementsData> {
```

Comment:
If you flag you function as `async` there is no point returning a `Promise` (It would de-sugare as Promise<Promise<...>>)